framework/src/NormalisedSumPDF.cpp

```diff
@@ -295,119 +308,85 @@ double NormalisedSumPDF::Normalisation( DataPoint* NewDataPoint, PhaseSpaceBound
 {
 	(void)NewDataPoint;
 	(void)NewBoundary;
+	// Calculate and cache the integrals
+	for(DataPoint* pdp: integrationBoundary->GetDiscreteCombinations())
```

Comment:
I wonder if this loop is actually necessary, or if it causes there to be N<sup>2</sup> integrals instead of N. `Normalisation()` takes a `DataPoint` and a `PhaseSpaceBoundary` as its arguments, so I would guess there might already be a loop over discrete combinations taking place outside the PDF. @rob-c  is this true?